tinygrad/tensor.py

```diff
@@ -37,6 +37,7 @@ class Tensor:
   def __init__(self, data:Union[int, float, list, tuple, LazyBuffer, np.ndarray], device=Device.DEFAULT, dtype:Optional[DType]=None, requires_grad:Optional[bool]=None):
     assert dtype is None or isinstance(dtype, DType), f"invalid dtype {dtype}"
     device = Device.canonicalize(device)
+    assert dtype is None or device.split(":", 1)[0] not in dtype.excluded, f"dtype: {dtype} not supported on device: {device}"
```

Comment:
These asserts should probably be at a much lower level, like when you create the rawbuffer. Then you only need one assert, and you don't need the excluded on dtypes either.